src/Lender.sol

```diff
@@ -599,7 +599,7 @@ contract Lender {
         // If the lst parameter is not populated, a swap is not required
         if (lst == address(0)) {
             // Conduct the lend operation to acquire principal tokens
-            (success, returndata) = _Market.adapters[p].delegatecall(
+            (success, returndata) = MarketPlace(marketplace).adapters(p).delegatecall(
```

Comment:
... and here you are using `MarketPlace(marketplace).adapters` (the contract)

Any specific reason for choosing 2 different versions of acquiring the adapters?

Reply:
It looks like it was included initially by sourabh for the `Principals` Enum, which I think I can include in the marketplace interface itself to work around that.